hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -118,7 +143,34 @@ private BaselineEntity getBaselineEntity(
                 attrMap,
                 containerMap,
                 incomingEdges,
-                outgoingEdges));
+                outgoingEdges,
+                labelResultSet));
+  }
+
+  private LabelResultSet buildLabelResultSet(
```

Comment:
Let me look in to this. We didn't go joiner route as we are not fetching label ids based on entities labels. We are fetching all and then filtering based on respective entity's labels.

Reply:
server-side vs in-memory filtering is a bit of an implementation detail - it's more about separating out the actual join glue (and deduping it since we'll want to join in other spots like explorer, too). Anyway, the big thing is keeping the server side of labels encapsulated within the label schema.